src/SonarScanner.MSBuild.PreProcessor/ProcessedArgs.cs

```diff
@@ -124,6 +129,14 @@ public string PropertiesFileName
             var isHostSet = AggregateProperties.TryGetValue(SonarProperties.HostUrl, out var sonarHostUrl); // Used for SQ and may also be set to https://SonarCloud.io
             var isSonarcloudSet = AggregateProperties.TryGetValue(SonarProperties.SonarcloudUrl, out var sonarcloudUrl);
             this.sonarServer = GetSonarServer(logger, isHostSet, sonarHostUrl, isSonarcloudSet, sonarcloudUrl);
```

Comment:
What do you think about failing early if the server is null? Do we have a way to successfully execute when this happens?

Reply:
We fail "early" a bit later. That part was added in #1967 here https://github.com/SonarSource/sonar-scanner-msbuild/pull/1967/files#diff-48770ce2f6a2f29edb2b82c1afb8de1b5970c3cc3bee83d12c5029a34ee85fbfR154